pkg/csi/cinder/openstack/openstack_backups.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package openstack snapshots provides an implementation of Cinder Snapshot features
+// cinder functions using Gophercloud.
+package openstack
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/extensions/backups"
+	"github.com/gophercloud/gophercloud/pagination"
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cloud-provider-openstack/pkg/metrics"
+	"k8s.io/klog/v2"
+)
+
+const (
+	backupReadyStatus                    = "available"
+	backupDescription                    = "Created by OpenStack Cinder CSI driver"
+	backupReadyDuration                  = 5 * time.Second
+	backupReadyFactor                    = 1.2
+	backupReadySteps                     = 15
+	backupMaxDurationSecondsPerGBDefault = 20
+	BackupMaxDurationPerGB               = "backup-max-duration-seconds-per-gb"
+)
+
```

Comment:
Could you add some comments on top of these public functions? Just like `openstack_instances.go` and `openstack_snapshots.go` did. So that `godoc` can auto-generate docs.